app/policies/user_policy.rb

```diff
@@ -34,4 +34,10 @@ def resolve
       scope
     end
   end
+
+  private
+
+  def draw_intent_state
+    !user.draw.intent_locked && !user.draw.draft? if user.draw
```

Comment:
Wait, looking below it looks like you did implement (lock intent when we start the lottery) - if that's the case, can't we replace `!user.draw.draft?` with `user.draw.pre_lottery?`? Admins will still be able to update intents.